File: tutorial_5/sample_1/main.py
import cv2 as cv
import numpy as np

from absl import app
from pathlib import Path


def main(argv):

    image = np.zeros((480, 640, 1))

    image = cv.circle(image, (320, 240), 50, (125), -1)

    ret, thresholded_image = cv.threshold(image, 124, 255, cv.THRESH_BINARY)

    cv.imshow("thresholded_image", thresholded_image)

    cv.waitKey(1)

    element_size = 10
    element = cv.getStructuringElement(cv.MORPH_ELLIPSE, (2 * element_size + 1, 2 * element_size + 1), (element_size, element_size))

    eroded_image = cv.erode(thresholded_image, element)

    cv.imshow("eroded_image", eroded_image)
    cv.waitKey(1)

    dilated_image = cv.dilate(thresholded_image, element)

    cv.imshow("dilated_imagte", dilated_image)
    cv.waitKey(1)

    cv.waitKey(0)

    cv.destroyAllWindows()

    return 0


if __name__ == "__main__":

    app.run(main)

File: tutorial_5/sample_1/main.cpp
#include "main.hpp"

int main(int argc, char* argv[]) {

	cv::Mat image{ cv::Mat(cv::Size(640, 480), CV_8UC1, cv::Scalar(0)) };

	cv::circle(image, cv::Point(320, 240), 50, cv::Scalar(125), -1);

	cv::Mat thresholdedImage;

	cv::threshold(image, thresholdedImage, 124, 255, cv::THRESH_BINARY);

	cv::imshow("thresholdedImage", thresholdedImage);

	cv::waitKey(1);

	int elementSize = 10;
	cv::Mat element = cv::getStructuringElement(cv::MORPH_ELLIPSE, cv::Size(2 * elementSize + 1, 2 * elementSize + 1), cv::Point(elementSize, elementSize));

	std::cout << element << std::endl;

	cv::Mat erodedImage;

	cv::erode(thresholdedImage, erodedImage, element);

	cv::imshow("erodedImage", erodedImage);
	cv::waitKey(1);

	cv::Mat dilatedImage;

	cv::dilate(thresholdedImage, dilatedImage, element);

	cv::imshow("dilatedImage", dilatedImage);
	cv::waitKey(1);

	cv::waitKey(0);

	cv::destroyAllWindows();

    return 0;
    
}